### PR TITLE
Manage Students: modify add api to include share setting

### PR DESCRIPTION
**Pull request by caleybrock**
Functions used for `/v2/sections/${sectionId}/students`

### CODE REVIEW

**Comment by caleybrock** [comment]
For the sake of time, I'm opting to update one of the APIs (Add students only used for word and picture sections), instead of create a whole new bulk API. To me, these were simple enough changes. @davidsbailey @Erin007 thoughts on this?

File: pegasus/helpers/section_api_helpers.rb [hunk 1]
```diff
@@ -33,6 +33,7 @@ def self.create(params)
         user_type: 'student',
         provider: 'sponsored',
         gender: gender,
+        properties: {sharing_disabled: sharing_disabled}.to_json,
```
Comment (Erin007) [inline pegasus/helpers/section_api_helpers.rb:36]: Why bury `sharing_disabled` in properties?

Comment (caleybrock) [reply pegasus/helpers/section_api_helpers.rb:36]: It currently lives in properties on the `User` model. 

This was a decision we made when we originally implemented this feature. I think we were to start from scratch with `User` we would put many more things on properties and not on the model directly. Or at least in a separate `Properties` model. 
History in some original PRs (tech spec linked): 
- https://github.com/code-dot-org/code-dot-org/pull/17064
- https://github.com/code-dot-org/code-dot-org/pull/17162

Comment (Erin007) [reply pegasus/helpers/section_api_helpers.rb:36]: Aha. That makes sense. Thanks for the context.

**Comment by caleybrock** [comment]
Cool, my concern was mostly that it would be better to rebuild in dashboard since that's our end goal. I'll check with Poorva/Brad and see if they are already tracking any tech debt for May 11, otherwise, I'll create something.

**Comment by caleybrock** [comment]
Actually, added a section to our * Labs tech debt doc.

**Pull request merged**